src/components/transformation/create/GitPodButton.tsx

```diff
@@ -66,30 +79,70 @@ function GitPodButton({
                     })
                 );
             }
-            const draft = await createEntityDraft(catalogName);
-            if (draft.error) {
-                throw new Error(
-                    `[${draft.error.code}]: ${draft.error.message}, ${draft.error.details}, ${draft.error.hint}`
+
+            let evaluatedDraftId = '';
+
+            if (draftId) {
+                evaluatedDraftId = draftId;
+
+                const existingMigrations = Object.values(migrations);
+                const existingTransforms = Object.values(transformConfigs);
+
+                const spec = generateInitialSpec(
+                    language,
+                    catalogName,
+                    sourceCollectionArray,
+                    {
+                        existingTransforms,
+                        existingMigrations,
+                        templateFiles: true,
+                    }
+                );
+
+                await modifyDraftSpec(spec, {
+                    draft_id: evaluatedDraftId,
+                    catalog_name: catalogName,
+                    spec_type: 'collection',
+                });
+            } else if (sourceCollectionSet) {
+                const draft = await createEntityDraft(catalogName);
+
+                if (draft.error) {
+                    throw new Error(
+                        `[${draft.error.code}]: ${draft.error.message}, ${draft.error.details}, ${draft.error.hint}`
+                    );
```

Comment:
Not for this PR - but I think we should handle the errors a bit different. Returning a string that contains all the props is nice but most of these are not truly usable. We should make this just handle errors the same way other create flows do.